web/e2e-tests/mention.test.ts

```diff
@@ -15,6 +15,12 @@ async function test_mention(page: Page): Promise<void> {
         stream_message_recipient_stream: "Verona",
         stream_message_recipient_topic: "Test mention all",
     });
+
+    // Ensure no typeahead or modals are opened.
+    await page.click("#compose-textarea");
+    await common.wait_for_modal_to_close(page);
+    await common.wait_for_micromodal_to_close(page);
+
```

Comment:
Could you explain why's this needed in this PR?

Reply:
There is some Puppeteer flakes where the typeahead doesn't close in time and ends up messing up with the mention typeahead. These are some patches to relieve the problems but seems like the flakes are still there with the failing test case.

Reply:
I see, these would be better in a separate commit (if we want to merge this)

Reply:
I'm a little confused, shouldn't testcases be in the same commit along with its changes? How come we should create a separate commit?